src/tool/yamlupgrade.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) rAthena Dev Teams - Licensed under GNU GPL
+// For more information, see LICENCE in the main folder
+
+#include "yamlupgrade.hpp"
+
+template<typename Func>
+bool process(const std::string &type, uint32 version, const std::vector<std::string> &paths, const std::string &name, Func lambda) {
+	for (const std::string &path : paths) {
+		const std::string name_ext = name + ".yml";
+		const std::string from = path + "/" + name_ext;
+		const std::string to = path + "/" + name + "-upgrade.yml";
+
+		inNode.reset();
+		inNode = YAML::LoadFile(from);
+
+		if (fileExists(from) && getHeaderVersion(inNode) < version) {
+			if (!askConfirmation("Found the file \"%s\", which requires an upgrade.\nDo you want to convert it now? (Y/N)\n", from.c_str())) {
+				continue;
+			}
+
+			if (fileExists(to)) {
+				if (!askConfirmation("The file \"%s\" already exists.\nDo you want to replace it? (Y/N)\n", to.c_str())) {
+					continue;
+				}
+			}
+
```

Comment:
Put some status output here so that the user knows processing will start now (always annoyed me for csv2yaml that you dont know if you pressed "Y" correctly...)